functions/src/modules/SQR.functions.ts

```diff
@@ -438,50 +395,40 @@ export const syncAllotments = functions.database
     ): Promise<any> => {
       const { listName, fileName } = context.params;
       const changedValues = change.after.val();
-      const spreadsheetId = functions.config().sqr.spreadsheet_id;
 
-      const gsheets = new GoogleSheet();
-      const allotmentFileNames = await gsheets.getColumn(
-        spreadsheetId,
-        ISoundQualityReportSheet.Allotments,
-        'File Name'
+      const gsheets = new GoogleSheets(
+        functions.config().sqr.spreadsheet_id,
+        ISoundQualityReportSheet.Allotments
       );
-
+      const allotmentFileNames = await gsheets.getColumn('File Name');
       const rowNumber = allotmentFileNames.indexOf(fileName) + 1;
       const { languages, notes, soundQualityReporting } = changedValues;
+
       const {
         timestampGiven,
         assignee,
         timestampDone,
         followUp,
       } = soundQualityReporting;
-      const updateResults = await gsheets.updateAllotmentRow(
-        spreadsheetId,
-        ISoundQualityReportSheet.Allotments,
-        rowNumber,
-        {
-          days_passed: EMPTY_VALUE,
-          date_given: spreadsheetDateFormat(timestampGiven),
-          notes: withDefault(notes),
-          language: commaSeparated(languages),
-          status: soundQualityReporting.status,
-          file_name: fileName,
-          devotee: withDefault(assignee.name),
-          email: withDefault(assignee.emailAddress),
-          phone: EMPTY_VALUE,
-          location: EMPTY_VALUE,
-          date_done: spreadsheetDateFormat(timestampDone),
-          follow_up: withDefault(followUp),
-          list: withDefault(listName),
-          serial: EMPTY_VALUE,
-        }
-      );
+
+      const updateResults = await gsheets.updateRow(rowNumber, {
```

Comment:
I think it's better to fetch the row, update some of its properties and then pass it to `updateRow`.

Reply:
You . mean like this?

```
      const existingRow: any = await gsheets.getRow(rowNumber);

      const updateResults = await gsheets.updateRow(rowNumber, {
        "Date Given": spreadsheetDateFormat(timestampGiven),
        "Notes": withDefault(notes),
        "Language": commaSeparated(languages),
        "Status": soundQualityReporting.status,
        "File Name": fileName,
        "Devotee": withDefault(assignee.name),
        "Email": withDefault(assignee.emailAddress),
        "Date Done": spreadsheetDateFormat(timestampDone),
        "Follow Up": withDefault(followUp),
        // Must be updated with existing value or it will be wiped by wrapper
        "List": existingRow["List"],
        "Serial": existingRow["Serial"],
      });
```

Reply:
Or maybe like this is cleaner?
```
      const existingRow: any = await gsheets.getRow(rowNumber);
      existingRow["Date Given"] = spreadsheetDateFormat(timestampGiven);
      existingRow["Notes"] = withDefault(notes);
      existingRow["Language"] = commaSeparated(languages);
      existingRow["Status"] = soundQualityReporting.status;
      existingRow["File Name"] = fileName;
      existingRow["Devotee"] = withDefault(assignee.name);
      existingRow["Email"] = withDefault(assignee.emailAddress);
      existingRow["Date Done"] = spreadsheetDateFormat(timestampDone);
      existingRow["Follow Up"] = withDefault(followUp);
      const updateResults = await gsheets.updateRow(rowNumber, existingRow);
```